engine/planner.go

```diff
@@ -268,8 +270,29 @@ func (p *planner) listBusy(ctx context.Context) (map[string]struct{}, error) {
 // helper function returns true if the os, arch, variant
 // and kernel match the stage.
 func (p *planner) match(stage *drone.Stage) bool {
+	labelMatch := true
+
+	if len(p.labels) > 0 || len(stage.Labels) > 0 {
+		labelMatch = p.matchLabel(stage)
+	}
+
 	return stage.OS == p.os &&
 		stage.Arch == p.arch &&
 		stage.Variant == p.version &&
-		stage.Kernel == p.kernel
+		stage.Kernel == p.kernel &&
+		labelMatch
+}
+
+func (p *planner) matchLabel(stage *drone.Stage) bool {
+	plannerLabels := formatLabels(p.labels)
+
+	for key, val := range stage.Labels {
+		if plannerVal, ok := plannerLabels[key]; ok {
+			if strings.EqualFold(plannerVal, val) {
+				return true
```

Comment:
this should return false if a single label does not match, and should return true only if all labels match.

Reply:
ok cool, ya i wasn't sure on that one. i'll use `reflect.DeepEqual` like @msaizar mentioned above

Reply:
I recommend copying the exact same function we are using to match on the server, to ensure the logic syncs up

https://github.com/drone/drone/blob/master/scheduler/queue/queue.go#L232
https://github.com/drone/drone/blob/master/scheduler/queue/queue.go#L175

Reply:
ah, ok. will do

Reply:
ok, i've got that updated